### DISCUSSION TITLE
Cypress detects visible element as invisible

### DISCUSSION THREAD

**Discussion by kgalushchak** [cypress-io/cypress#27700] [Questions and Help] [answered]
How does Cypress detect if element is visible? I assumed it uses element's `display` property, though, it feels like something else matters.
I have a test where I try to locate an element which is visible in the page (I can see it there), although Cypress detects it as invisible.

<img width="507" alt="Screenshot 2023-08-30 at 12 22 47" src="https://github.com/cypress-io/cypress/assets/82284603/f4f84727-5ec2-4911-bc8b-03add668725d">

`display` of my attribute is not `none`:

![display](https://github.com/cypress-io/cypress/assets/82284603/fe139ce9-6845-46f2-8b6d-a0bd466e476e)

Can someone advise how to get Cypress to see this element?

UPD. This is an element which is displayed for ~5 seconds after a click on another element and then disappears, though, I clearly see it is visible when Cypress tries to locate it when running tests in Cypress runner.

**Comment by michallbanas** [comment, marked as answer]
Hey hey. You can find how Cypress checks visibility here: https://docs.cypress.io/guides/core-concepts/interacting-with-elements#Visibility

But based on the screenshot, the test failed due to contain assertion, or?

**Reply by kgalushchak** [reply]
Thank you! The link is very useful, it explains what else is considered when checking if element is visible.
